### PR TITLE
Adding font Size as default parameter

### PR DESCRIPTION
**Pull request by saikarna913**
Adding the font size as default parameter
This PR Solves #29202 

## PR summary

-This PR automatically adjusts the font size when creating the table, provided the fontsize argument is specified, without requiring an explicit call to the set_fontsize function.
-Automatically applying the font size ensures consistent styling during table creation, streamlining the process for users. It eliminates the need for additional method calls

## PR checklist

- [x] closes fontsize in tables not working #29202 
- [x]  added the type hint and ran the boilerplate
- [x] tested the code with a new test case added in test_table.py
Test code:
```
import matplotlib.pyplot as plt
import numpy as np

# Data for plotting
x = np.linspace(0, 10, 100)
y = x + 1
tableData = [['a', 1], ['b', 1]]

# Create the figure and axis objects
fig, ax = plt.subplots()

# Plot the data
ax.plot(x, y)

# Add a table
t = ax.table(
    cellText=tableData,  # Table content
    loc='top',  # Position of the table
    cellLoc='center',  # Center-align the text in cells
    fontsize=30  # Set the font size for the table
)

# Save the figure as an image
plt.savefig('output_plot_with_table_after.png')  # Save as PNG file

# Optionally, close the plot (no need to display it)
plt.close()
```
I got this output
image

### CODE REVIEW

**Comment by saikarna913** [comment]
@tacaswell please help me why mypy is failing I haven't changed the pylab.py code

**Comment by timhoffm** [comment]
The mypy issue is unrelated and happens also on other PRs. It can be ignored here.

**Comment by saikarna913** [comment]
@rcomer  can you review my PR and approve it, As @timhoffm suggestion mypy issue can be ignored I guess.

**Comment by saikarna913** [comment]
@timhoffm i guess now this PR can be merged. it would be a great encouragement for me as a beginner in open source. if there is any issue please let me know

**Comment by saikarna913** [comment]
@timhoffm I don't understand why these are passing sometimes and failing. in the last commit pre-commit. ci failed due to the whitespace, so I removed the whitespace and then [Tests / Python 3.13 on macos-14 (pull_request)] is failing

**Comment by rcomer** [comment]
Unfortunately the MacOS tests can be flakey and sometimes show failures that are not caused by the PR.  You just got unlucky when you removed the white space.

File: lib/matplotlib/table.py [hunk 1]
```diff
@@ -839,4 +839,7 @@ def table(ax,
             table.auto_set_column_width(-1)
 
     ax.add_table(table)
+    # Set the fontsize if provided
+    if fontsize is not None:
+        table.set_fontsize(fontsize)
```
Comment (rcomer) [inline lib/matplotlib/table.py:844]: I suggest putting this before `add_table` because it is part of defining the table so belongs with the cell adding logic.  Also it would be worth expanding the comment to explain _why_ we do this here rather than leaving it as part of kwargs.  Something like

`# set_fontsize is only effective after cells are added`

File: lib/matplotlib/table.pyi [hunk 2]
```diff
@@ -8,12 +8,14 @@ from .transforms import Bbox
 from .typing import ColorType
 
 from collections.abc import Sequence
-from typing import Any, Literal, TYPE_CHECKING
+from typing import Any, Literal
 
 from pandas import DataFrame
 
+
```
Comment (rcomer) [inline lib/matplotlib/table.pyi:15]: Please do not add blank lines that are unrelated to you actual code changes.

File: lib/matplotlib/tests/test_table.py [hunk 3]
```diff
@@ -270,3 +270,24 @@ def test_table_dataframe(pd):
...
+
+
+# Test function for fontsize in table
+def test_table_fontsize():
+    # Data for plotting
+    tableData = [['a', 1], ['b', 1]]
+    # Create the figure and axis objects
+    fig, ax = plt.subplots()
+    # Plot the data (although we are focusing on the table)
+    ax.plot(np.linspace(0, 10, 100), np.linspace(0, 10, 100) + 1)
```
Comment (rcomer) [inline lib/matplotlib/tests/test_table.py:282]: This appears to be plotting a straight line that is unrelated to the table.  Can you explain why it is here?  If not, take it out.

File: lib/matplotlib/tests/test_table.py [hunk 4]
```diff
@@ -270,3 +270,24 @@ def test_table_dataframe(pd):
     for r, (index, row) in enumerate(df.iterrows()):
         for c, col in enumerate(df.columns if r == 0 else row.values):
             assert table[r if r == 0 else r+1, c].get_text().get_text() == str(col)
+
+
+# Test function for fontsize in table
```
Comment (rcomer) [inline lib/matplotlib/tests/test_table.py:275]: I would move this comment inside the test and make it more explicit e.g. "Test that the passed fontsize propagates to cells".  I also think the comments currently in the test are unnecessary as they are just stating what the code does, which someone familiar with Matplotlib can tell by reading the code itself.

File: lib/matplotlib/table.pyi (hunk 2 above)
Comment (saikarna913) [reply lib/matplotlib/table.pyi:15]: it is added to remove the flake issue

File: lib/matplotlib/table.py [hunk 5]
```diff
@@ -838,5 +838,8 @@ def table(ax,
         if rowLabelWidth == 0:
             table.auto_set_column_width(-1)
 
+    # set_fontsize is only effective after cells are added
+    if fontsize is not None:
+        table.set_fontsize(fontsize)
```
Comment (rcomer) [inline lib/matplotlib/table.py:843]: Add a blank line after this.  It will make it clear to the reader that adding the table is a separate step.

File: lib/matplotlib/table.pyi (hunk 2 above)
Comment (rcomer) [reply lib/matplotlib/table.pyi:15]: I think we do not usually apply `flake8` to `.pyi` files but am happy if anyone can confirm or correct me on that.

**Review by rcomer — COMMENTED** [review]
Thanks @saikarna913 this looks good to me and my only remaining comments are about blank lines.

**Review by rcomer — APPROVED** [review]

**Review by oscargus — CHANGES_REQUESTED** [review]
Please update the doc-string to reflect the new parameter. Also, add ``.. versionadded:: 3.11`` there.

**Comment by rcomer** [comment]
I wasn’t sure if a docstring change would be needed since _fontsize_ is already listed as a `Table` property under kwargs.

**Comment by timhoffm** [comment]
> I wasn’t sure if a docstring change would be needed since _fontsize_ is already listed as a `Table` property under kwargs.

The important thing is that every parameter is documented, direct or indirectly. There is no 1:1 relation that an explicit parameter must be documented explicitly, or vice versa. Typically, if we use explicit parameters for Artist properties, it's because they have a special meaning / modified functionality, which warrant explicit documentation. But here we just need the value to make the table behave as expected. We don't need to communicate additional information and thus don't need an explicit docstring.

File: lib/matplotlib/table.py [hunk 6]
```diff
@@ -653,7 +653,7 @@ def table(ax,
           cellLoc='right', colWidths=None,
           rowLabels=None, rowColours=None, rowLoc='left',
           colLabels=None, colColours=None, colLoc='center',
-          loc='bottom', bbox=None, edges='closed',
+          loc='bottom', bbox=None, edges='closed', fontsize=None,
```
Comment (timhoffm) [inline lib/matplotlib/table.py:656]:
```suggestion
          loc='bottom', bbox=None, edges='closed', *, fontsize=None,
```
Let's not accept this as positional.

The alternative to making `fontsize` an explicit parameter and changing all related signatures would be to extract it from kwargs. That would leave all signatures literally unchanged.

File: lib/matplotlib/table.py [hunk 7]
```diff
@@ -838,5 +838,9 @@ def table(ax,
         if rowLabelWidth == 0:
             table.auto_set_column_width(-1)
 
+    # set_fontsize is only effective after cells are added
+    if fontsize is not None:
+        table.set_fontsize(fontsize)
```
Comment (timhoffm) [inline lib/matplotlib/table.py:843]: Optional: As an alternative to changing the spelling of our signatures, we could get fontsize from kwargs here. That way, it doesn't show up in the signature explicitly (which may be an advantage or disavantage depending on your perspective.

```suggestion
    if "fontsize" in kwargs:
        table.set_fontsize(kwargs["fontsize"])
```

**Review by timhoffm — COMMENTED** [review]
Take one of my two comments: Either make the new `fontsize` parameter keyword-only or don't introduce an explicit `fontsize` parameter and instead take the value from kwargs.

**Comment by oscargus** [comment]
Thanks! One may consider if this is a bugfix and therefore should be backported.

**Comment by timhoffm** [comment]
@meeseeksdev backport to v3.10.1

**Comment by timhoffm** [comment]
@meeseeksdev backport to v3.10.x
